tests/integration/traces/test_traces_command.py

```diff
@@ -0,0 +1,223 @@
+import time
+from pathlib import Path
+from unittest import skipIf
+
+import boto3
+from tests.integration.deploy.deploy_integ_base import DeployIntegBase
+from tests.integration.package.package_integ_base import PackageIntegBase
+from tests.integration.traces.traces_integ_base import TracesIntegBase, RETRY_COUNT
+from tests.testing_utils import run_command, RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+from datetime import datetime
+import logging
+from parameterized import parameterized
+
+LOG = logging.getLogger(__name__)
+
+SKIP_TRACES_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+
+
+@skipIf(SKIP_TRACES_TESTS, "Skip traces tests in CI/CD only")
+class TestTracesCommand(TracesIntegBase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.lambda_client = boto3.client("lambda")
+        cls.cfn_client = boto3.client("cloudformation")
+        cls.xray_client = boto3.client("xray")
+
+        cls.test_data_path = Path(__file__).resolve().parents[1].joinpath("testdata", "traces")
+
+        package = PackageIntegBase()
+        cls.stack_name = package._method_to_stack_name("test.integration.test_traces_command")
+
+        LOG.info("Deploying stack %s", cls.stack_name)
+        deploy = DeployIntegBase()
+        deploy_cmd = deploy.get_deploy_command_list(
+            stack_name=cls.stack_name,
+            template_file=cls.test_data_path.joinpath("python-apigw-sfn", "template.yaml"),
+            resolve_s3=True,
+            capabilities="CAPABILITY_IAM",
+        )
+        deploy_result = run_command(deploy_cmd)
+
+        cls.stack_resources = cls.cfn_client.describe_stack_resources(
+            StackName=cls.stack_name
+        ).get(
+            "StackResources", []
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.cfn_client.delete_stack(StackName=cls.stack_name)
```

Comment:
Will this leave any resource leaking in the account? I remember deploy and sync tests have more complicated cleanup steps, wonder if this would be different or not.

Reply:
Sync and deploy create stacks in each test case, but in trace we only create stack once in the `setUpClass` then use it for all the rest test cases. So we just need to delete 1 stack in the 'tearDownClass' after all tests are done.